src/app/App.js

```diff
@@ -1,31 +1,63 @@
 /** @flow
  * @format */
+
+import '../globals';
+
 import React from 'react';
 
+// Gutenberg imports
+import { setLocaleData } from '@wordpress/i18n';
+
 import AppContainer from './AppContainer';
 import initialHtml from './initial-html';
 
+import { getTranslation } from '../../i18n-cache';
+
 type PropsType = {
 	initialData: string,
 	initialHtmlModeEnabled: boolean,
 	initialTitle: string,
+	locale: string,
+	translations: mixed,
 };
 
-const AppProvider = ( { initialTitle, initialData, initialHtmlModeEnabled }: PropsType ) => {
-	if ( initialData === undefined ) {
-		initialData = initialHtml;
+export default class AppProvider extends React.Component<PropsType> {
+	constructor( props: PropsType ) {
+		super( props );
+
+		this.setLocale( props.locale );
 	}
 
-	if ( initialTitle === undefined ) {
-		initialTitle = 'Welcome to Gutenberg!';
+	componentDidUpdate( prevProps: PropsType ) {
+		if ( prevProps.locale !== this.props.locale ) {
+			this.setLocale( this.props.locale );
+		}
 	}
 
-	return (
-		<AppContainer
-			initialHtml={ initialData }
-			initialHtmlModeEnabled={ initialHtmlModeEnabled }
-			initialTitle={ initialTitle } />
-	);
-};
+	setLocale( locale: ?string ) {
+		const translationsFromParentApp = this.props.translations || {};
+		const translations = Object.assign( {}, translationsFromParentApp, getTranslation( locale ) );
```

Comment:
Hmm, I'm thinking, how about giving the WP apps priority here? Like, if there's a conflict between what the parent sends and what Gutenberg has then let's give the parent priority. That would also give us the ability to have each mobile platform decide to overwrite a string, for whatever reason (e.g. platform guidelines). WDYT?

Reply:
It sounds sane. The only thing I'm afraid of though is mixing english and an other language in the case of using a language which is supported by the app but not by gutenberg for instance. 

If you don't think it's a problem I can give the parent app priority

Reply:
Hmm, won't that be true the other way too? Like, if some strings are translated by GB but the native specific ones are not, right?

Reply:
Yeah, I went with your approach in the end and decided to mix both and give priority to the parent app 👍

Reply:
All in all, I think that giving priority to the native side strings should be a good start. We can revise if we see it be problematic.

Reply:
I checked the current Spanish translations in both apps and Gutenberg for common strings, and everything seems pretty consistent. The only difference I've spotted that's perhaps relevant is how media library is translated, and it makes sense that we're consistent with the app translation, so good call 😁